app/services/import_polygons.rb

```diff
@@ -65,6 +46,28 @@ def location_in_scope?
       api_location_type == :cities && DOWNCASE_ONS_CITIES.include?(location_name)
   end
 
+  def transform_geometry_rings_to_polygons(geometry_rings)
+    # When the ONS API returns multiple 'rings' for a location, some of these are small islands or similar. Some of
+    # these islands are inhabited and have a school on, and others do not. (E.g. Mersea Island in Essex, Hayling Island
+    # in Hampshire, Sheppey Island in Kent, St Mary's in the Isles of Scilly.)
+    #
+    # Store the 'rings' in a hash because Postgres doesn't support multidimensional arrays of varying extents.
+    #
+    # If algolia searches by polygon are slow, (some of) these boundaries could be downsampled significantly.
+
+    polygons = []
```

Comment:
While you're here (although hopefully we'll get to kill all this next sprint): Both the outer and inner loop should really be using `map` instead of creating empty arrays and pushing into them!